### PR TITLE
Multitarget wrapper should build with Asserts enabled

### PR DESCRIPTION
**Pull request by steven-johnson**
Building with NoAsserts doesn’t allow propagation of nonzero results.
Also add a test for this to multitarget_aottest.cpp

### CODE REVIEW

File: test/generator/multitarget_aottest.cpp [hunk 1]
```diff
@@ -7,6 +7,11 @@
 
 using namespace Halide::Tools;
 
+// Override halide_error() so that an intentional error doesn't trigger exit()
+extern "C" void halide_error(void *user_context, const char *message) {
```
Comment (abadams) [inline test/generator/multitarget_aottest.cpp]: Weak-linkage overloading doesn't work on windows/mingw. Use halide_set_error_handler instead.

Comment (steven-johnson) [inline test/generator/multitarget_aottest.cpp]: done

File: test/generator/multitarget_aottest.cpp [hunk 2]
```diff
@@ -7,6 +7,10 @@
 
 using namespace Halide::Tools;
 
+void my_error_handler(void *user_context, const char *message) {
+  printf("Saw Error: (%s)\n", message);
```
Comment (abadams) [inline test/generator/multitarget_aottest.cpp]: nit: four-space indentation

Comment (steven-johnson) [inline test/generator/multitarget_aottest.cpp]: done

**Pull request merged**